File: recursion/s8-problems-1/exercise-10/solution.go
package main

import "fmt"

func main() {
	fmt.Println(power(2, 0))
	fmt.Println(power(2, 2))
	fmt.Println(power(2, 4))
}

func power(base, exp int) int {
	if exp == 0 {
		return 1
	}
	if exp == 1 {
		return base
	}
	return base * power(base, exp-1)
}

File: recursion/s8-problems-1/exercise-10/solution.js
/*
Write a function called power which accepts a base and an exponent.
The function should return the power of the base to the exponent.
This function should mimic the functionality of Math.power() - do not worry about negative bases and exponents.
*/

const power = function (base, exp) {
  if (exp === 0) return 1;
  if (exp === 1) return base;
  return base * power(base, exp - 1);
};

console.log(power(2, 0)); // 1
console.log(power(2, 2)); // 4
console.log(power(2, 4)); // 16
